lib/js-yaml/dumper.js

```diff
@@ -201,8 +201,9 @@ function isPlainSafe(c) {
     && c !== CHAR_LEFT_CURLY_BRACKET
     && c !== CHAR_RIGHT_CURLY_BRACKET
     // - ":" - "#"
+    // /* An ns-char preceding */ "#"
     && c !== CHAR_COLON
-    && c !== CHAR_SHARP;
+    && (c !== CHAR_SHARP || prev && isPrintable(prev));
 }
```

Comment:
`((c !== CHAR_SHARP) || (prev && isPrintable(prev)));`

I'd use more scopes to simplify understanding.